protocols/virtualfund/virtualfund_single_hop_test.go

```diff
@@ -0,0 +1,304 @@
+				outcome.Allocation{
+					Destination: alice.destination,
+					Amount:      big.NewInt(6),
+				},
+				outcome.Allocation{
+					Destination: bob.destination,
+					Amount:      big.NewInt(4),
+				},
+			},
+		}},
+		TurnNum: 0,
+		IsFinal: false,
+	}
+	var vPostFund = vPreFund.Clone()
+	vPostFund.TurnNum = 1
+
+	ledgers := make(map[types.Destination]actorLedgers)
+	ledgers[alice.destination] = actorLedgers{
+		right: prepareConsensusChannel(uint(consensus_channel.Leader), alice, p1),
+	}
+	ledgers[p1.destination] = actorLedgers{
+		left:  prepareConsensusChannel(uint(consensus_channel.Follower), alice, p1),
+		right: prepareConsensusChannel(uint(consensus_channel.Leader), p1, bob),
+	}
+	ledgers[bob.destination] = actorLedgers{
+		left: prepareConsensusChannel(uint(consensus_channel.Follower), p1, bob),
+	}
+
+	return testData{vPreFund, vPostFund, ledgers}
+}
+
+type Tester func(t *testing.T)
+
+func testNew(a actor) Tester {
+	return func(t *testing.T) {
+		td := newTestData()
+		lookup := td.ledgers
+		vPreFund := td.vPreFund
+
+		// Assert that a valid set of constructor args does not result in an error
+		o, err := constructFromState(
+			false,
+			vPreFund,
+			a.address,
+			lookup[a.destination].left,
+			lookup[a.destination].right,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		switch a.role {
+		case alice.role:
+			assert(t, o.ToMyLeft == nil, "left connection should be nil")
+			assert(t, diffFromCorrectConnection(o.ToMyRight, alice, p1) == "", "incorrect connection")
+		case p1.role:
+			assert(t, diffFromCorrectConnection(o.ToMyLeft, alice, p1) == "", "incorrect connection")
+			assert(t, diffFromCorrectConnection(o.ToMyRight, p1, bob) == "", "incorrect connection")
+		case bob.role:
+			assert(t, diffFromCorrectConnection(o.ToMyLeft, p1, bob) == "", "incorrect connection")
+			assert(t, o.ToMyRight == nil, "right connection should be nil")
+		}
+	}
+}
+
+// diffFromCorrectConnection compares the guarantee stored on a connection with
+// the guarantee we expect, given the expected left and right actors
+func diffFromCorrectConnection(c *Connection, left, right actor) string {
+	td := newTestData()
+	vPreFund := td.vPreFund
+
+	Id, _ := vPreFund.FixedPart().ChannelId()
+
+	// HACK: This should really be comparing GuaranteeInfo, but GuaranteeInfo
+	// contains types.Funds amounts in their LeftAmount and RightAmount fields.
+	// I am not sure how these types are meant to be used, and am
+	// comparing the _guarantees_ that we expect to include, instead of the GuaranteeInfo
+
+	expectedAmount := big.NewInt(0).Set(vPreFund.VariablePart().Outcome[0].TotalAllocated())
+	want := consensus_channel.NewGuarantee(expectedAmount, Id, left.destination, right.destination)
+	got := c.getExpectedGuarantee()
+
+	return compareGuarantees(want, got)
+}
+
+func TestNew(t *testing.T) {
+	for _, a := range allActors {
+		msg := fmt.Sprintf("Testing new as %v", a.name)
+		t.Run(msg, testNew(a))
+	}
+}
+
+func testClone(my actor) Tester {
```

Comment:
Could we rename this so it's distinct from `TestClone`? Maybe `testCloneAs`?

Reply:
c3b855de620e1f235ae2718acd5adcba29b33b97